libdnf5/utils/iniparser.cpp

```diff
@@ -25,6 +25,53 @@ namespace libdnf5 {
 
 constexpr char DELIMITER = '\n';
 
+namespace {
+
+// Returns the position of the first undoubled character ']'.
+std::size_t find_end_of_section_name(const std::string & str, std::size_t pos) {
+    if (pos >= str.size()) {
+        return std::string::npos;
+    }
+
+    bool bracket = false;
+    for (std::size_t idx = pos;; ++idx) {
+        const auto ch = str[idx];
+        if (bracket) {
+            if (ch != ']') {
+                return idx - 1;
+            }
+            bracket = false;
+        } else {
+            if (ch == '\0' || ch == '\n' || ch == '\r') {
+                return std::string::npos;
+            }
+            if (ch == ']') {
+                bracket = true;
+            }
+        }
+    }
+}
+
+// Returns the input string with the undoubled characters ']'.
+// A sequence of two chars ']' is returned as a single char ']'.
+std::string section_name_decode_bracket(const std::string & section_name) {
```

Comment:
I guess it is required for glob support in section header like [fedora[2-8]]. @jrohel Is it correct?

Reply:
What about to use the sema logic like we have in NEVRA parser - when the second `[` then the parser skips everything until the next `]`. What do you think?

Reply:
@j-mracek OK. I changed the logic to be compatible with the NEVRA parser.